### ISSUE TITLE
Failure of terraform deployments with `retention_policy` block

### ISSUE THREAD

**Issue by stefpiatek** [UCLH-Foundry/FlowEHR#320]
**Describe the bug**

Encountered this error during building AzureTre and expect it applies here too.

When building any new resources that have a `retention_policy` in their diagnostic settings there is now an error returned from azure:

```
Message="Diagnostic settings does not support retention for new diagnostic settings."
```

It looks like this was due for deprication at the end of [september 2023](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/migrate-to-azure-storage-lifecycle-policy), so seems like this has happened early. Existing resources _should_ be fine until september 2025

**To Reproduce**
1. Clone the repository
2. Create a fresh deployment
3. When a new resource is created with a retention policy, the terraform build will fail

**Expected behavior**
Should pass build and use the [Azure Storage lifecycle policy](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/migrate-to-azure-storage-lifecycle-policy) instead

**Additional context**
Seems like it started [yesterday](https://github.com/microsoft/AzureTRE/issues/3686)

**Comment by stefpiatek** [comment]
suggestion that removing the block entirely is the simplest option, would then default to the log analytics retention policy. @marrobi to confirm whether this has been set

**Comment by marrobi** [comment]
Log analytics rete is set to 90 days in FlowEHR, and 30 days in AzureTRE.

**Comment by stefpiatek** [comment]
Closed by #328
